src/service/presence/mod.rs

```diff
@@ -170,6 +170,44 @@ impl Service {
 		self.db.remove_presence(user_id).await;
 	}
 
+	// Unset online/unavailable presence to offline on startup
+	pub async fn unset_all_presence(&self) -> Result<()> {
+		for user_id in &self
+			.services
+			.users
+			.list_local_users()
+			.map(UserId::to_owned)
+			.collect::<Vec<_>>()
+			.await
+		{
+			let presence = self.db.get_presence(user_id).await;
+
+			let presence = match presence {
+				| Ok((_, ref presence)) => &presence.content,
+				| _ => return Ok(()),
```

Comment:
This also needs to be a `continue`

Reply:
I've fixed it locally and intend on pushing to your PR soon.

Reply:
Ok. So sorry for the delay.